webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -620,60 +620,100 @@ private static void addScrambles(PdfWriter docWriter, Document doc, ScrambleRequ
-        try {
-            ColumnText ct = null;
-            int status = 0;
-            if (maxFontSize <= 0) {
-                int cr = 0;
-                int lf = 0;
-                char[] t = text.toCharArray();
-                for (int k = 0; k < t.length; ++k) {
-                    if (t[k] == '\n') {
-                        ++lf;
-                    } else if (t[k] == '\r') {
-                        ++cr;
-                    }
-                }
-                int minLines = Math.max(cr, lf) + 1;
-                maxFontSize = Math.abs(rect.getHeight()) / minLines - 0.001f;
-            }
-            font.setSize(maxFontSize);
-            Phrase ph = new Phrase(text, font);
-            ct = new ColumnText(null);
-            ct.setSimpleColumn(ph, rect.getLeft(), rect.getBottom(), rect.getRight(), rect.getTop(), maxFontSize, Element.ALIGN_LEFT);
-            ct.setRunDirection(runDirection);
-            status = ct.go(true);
-            if ((status & ColumnText.NO_MORE_TEXT) != 0 && (newlinesAllowed || ct.getLinesWritten() <= 1)) {
-                return maxFontSize;
-            }
+
+        //first verify if there's a line break ("\n") -> megaminx
+        //if so, calculate the font based only on the width
+        int lineBreak = text.indexOf("\n");
+
+        if(lineBreak > 0) {
+            //the char width is fontSize * 0.6 -> this was found with getWidthPoint()
+            float size = rect.getWidth() / (lineBreak * 0.6f);
+
+            //the calculated size is a little too big, so we reduce it to 95%
+            return (size * 0.95f);
+        }
+
+        //if we can use more than one line,
+        if(newlinesAllowed) {
+            float minFont = 8.0f;
+
+            float maxFont = maxFontSize;
+
             float precision = 0.1f;
-            float min = 0;
-            float max = maxFontSize;
-            float size = maxFontSize;
-            for (int k = 0; k < 50; ++k) { //just in case it doesn't converge
-                size = (min + max) / 2;
-                ct = new ColumnText(null);
-                font.setSize(size);
-                ct.setSimpleColumn(new Phrase(text, font), rect.getLeft(), rect.getBottom(), rect.getRight(), rect.getTop(), size, Element.ALIGN_LEFT);
-                ct.setRunDirection(runDirection);
-                status = ct.go(true);
-                if ((status & ColumnText.NO_MORE_TEXT) != 0 && (newlinesAllowed || ct.getLinesWritten() <= 1)) {
-                    if (max - min < size * precision) {
-                        return size;
-                    }
-                    min = size;
+
+            //first make the chunks using a potentialFontSize
+            //starting with the maximum size
+            float oldFont = minFont;
+            float newFont = maxFont;
+
+            //determine line height - using font ascent and descent
```

Comment:
Could you add a link to where you got the arithmetic from? It's still not clear to me why it would work.